src/apphosting/githubConnections.ts

```diff
@@ -90,13 +90,26 @@ const ADD_ACCOUNT_CHOICE = "@ADD_ACCOUNT";
 const MANAGE_INSTALLATION_CHOICE = "@MANAGE_INSTALLATION";
 
 /**
- * Prompts the user to link their backend to a GitHub repository.
+ * Prompts the user to create a GitHub connection.
  */
-export async function linkGitHubRepository(
+export async function getOrCreateGithubConnection(
   projectId: string,
   location: string,
-): Promise<devConnect.GitRepositoryLink> {
+  createConnectionId?: string,
+): Promise<devConnect.Connection> {
   utils.logBullet(clc.bold(`${clc.yellow("===")} Import a GitHub repository`));
+
+  if (createConnectionId) {
+    // Check if the connection already exists.
+    try {
+      const connection = await devConnect.getConnection(projectId, location, createConnectionId);
+      utils.logBullet(`Reusing existing connection ${createConnectionId}`);
+      return connection;
+    } catch (e) {
```

Comment:
It may be good to verify that this is a 404 error and continue to throw if not. We do this in other places as well, see: 
- https://github.com/firebase/firebase-tools/blob/master/src/apphosting/githubConnections.ts#L316
- https://github.com/firebase/firebase-tools/blob/master/src/apphosting/githubConnections.ts#L502